src/bin/purge_ttl.rs

```diff
@@ -298,34 +308,69 @@ fn main() -> Result<(), Box<dyn Error>> {
         let _timer_total = start_timer(&statsd, "purge_ttl.total_duration");
         {
             let _timer_batches = start_timer(&statsd, "purge_ttl.batches_duration");
-
-            if incremental {
-                delete_incremental(
-                    &client,
-                    &session,
-                    "batches".to_owned(),
-                    "batch_id".to_owned(),
-                    chunk_size,
-                    max_to_delete,
-                )?;
-            } else {
-                delete_all(&client, &session, "batches".to_owned())?;
+            let mut success = false;
+            for i in 0..retries {
+                match if incremental {
+                    delete_incremental(
+                        &client,
+                        &session,
+                        "batches".to_owned(),
+                        "batch_id".to_owned(),
+                        chunk_size,
+                        max_to_delete,
+                    )
+                } else {
+                    delete_all(&client, &session, "batches".to_owned())
+                } {
+                    Ok(_) => {
+                        success = true;
+                        break;
+                    }
+                    Err(e) => {
+                        warn!("Batch transaction error: {}: {:?}", i, e);
+                        if nap_time.as_secs() > 0 {
```

Comment:
any millis value < 1sec would produce 0 for `as_secs`:

```suggestion
                        if nap_time.as_millis() > 0 {
```

(and same below too)